providers/azure/azure.go

```diff
@@ -63,6 +63,9 @@ type Config struct {
 
 	// Deprecated: Is automatically set by the Azure SDK.
 	MSIResource string `yaml:"msi_resource"`
+
+	// IsAzureDataLakeGen2 indicates whether the provided storage account is an Azure Data Lake Gen2 account.
+	IsAzureDataLakeGen2 bool `yaml:"is_azure_data_lake_gen2"`
```

Comment:
Can this be interrogated from the account at runtime, rather than relying on user configuration?

Reply:
exactly right, the configuration might not be required

Reply:
@ringerc i pushed a commit about auto-detection https://github.com/thanos-io/objstore/pull/238/commits/3efbe73d21973ee4585dbcf572c4e4997e928c23

Reply:
@Anhui-tqhuang Damn - I'd (mostly) implemented in https://github.com/ringerc/thanos-objstore-client-patches/commit/559920163d6dc333b26d950d75117ab2986e291e but didn't get time to share it with you before EoD. Sorry for the overlap. I didn't get time to create storage accounts to test it against to verify whether I had to check `ContainerClientGetAccountInfoResponse.AccountKind` or `ContainerClientGetAccountInfoResponse.IsHierarchicalNamespaceEnabled`.

Take a look at my patch; you might want to borrow the docs update and the support for optionally overriding autodetection. I haven't had time to set up accounts to test the change though.

Reply:
@ringerc updated and merged some your change https://github.com/thanos-io/objstore/pull/238/commits/e1a02756456eec4123d2acb34c913d92495a06ca

moreover, i assume storageV2 is not equal to gen2, what we care more about is whether `IsHierarchicalNamespaceEnabled` is set to `true` or not, so i have some mutations

Reply:
Thanks; I wasn't sure about that